### PR TITLE
Java 11 upgrade

### PR DESCRIPTION
**Pull request by prathasirisha**
I left my comments on some of the changes to add clarity. 

Created this PR to start discussions on the changes, but we should merge them only after we cut a milestone release (as suggested by @nikhilnanivadekar )

### CODE REVIEW

File: pom.xml
Comment (prathasirisha) [inline pom.xml:560]: I could not find a better solution after spending some time on this. I am open to alternate solutions if there are any. Without this change, the builds fail with 
`Exit code: 2 - Loading source files for package org.eclipse.collections.codegenerator.model...
error: No source files for package org.eclipse.collections.codegenerator.model`

I tried a few different solutions suggested [here](https://stackoverflow.com/questions/58836862/jdk-11-and-javadoc) and this is the only one that seems to work. I added a new upload artifact step in javadoc workflow to test that it actually generates javadoc with this change.

File: pom.xml
Comment (prathasirisha) [inline pom.xml:572]: `excludePackageNames` doesn't seem to take as there is javadoc created for these excluded package names. I can revert this change, doesn't matter. I wanted to bring this to attention.

File: pom.xml
Comment (prathasirisha) [inline pom.xml:625]: Reference: https://maven.apache.org/plugins/maven-javadoc-plugin/javadoc-mojo.html#links

File: missing-dependency/missing-dependency.iml
Comment (prathasirisha) [inline missing-dependency/missing-dependency.iml:3]: I made a change to this file, but I am not entirely sure what this module does. @motlin do we need this, it seems this was committed in 2018 with just this iml file. Happy to revert this change and take it up later.

File: .github/workflows/javadoc.yml
Comment (prathasirisha) [inline .github/workflows/javadoc.yml:37]: I added this in case anyone wants to see the generated javadoc given this PR has related javadoc plugin changes. I will revert this section before the PR gets merged.

File: eclipse-collections-api/pom.xml
Comment (donraab) [inline eclipse-collections-api/pom.xml:91]: Why was this changed from 8 to 17 instead of 8 to 11? Was this because we are building Javadoc with 17?

Comment (donraab) [reply eclipse-collections-api/pom.xml:91]: > Why was this changed from 8 to 17 instead of 8 to 11? Was this because we are building Javadoc with 17?

Never mind @prathasirisha, I see you added a comment above referencing this change.

Comment (prathasirisha) [reply eclipse-collections-api/pom.xml:91]: I meant to reference Java 11 docs, looks like committed the changes before I could make that change. I will update them shortly.

**Review by donraab — APPROVED** [review]
This PR looks good to me. Thanks for contributing @prathasirisha! I would like to give time to hear any feedback and/or approvals from @nikhilnanivadekar and @motlin.

**Comment by nikhilnanivadekar** [comment]
Will take a look tomorrow. Please do not merge till milestone is cut.

**Comment by nikhilnanivadekar** [comment]
Milestone is published, will reflect in maven central as per the usual SLA. Will do a review later today/tomorrow.

**Comment by prathasirisha** [comment]
> Milestone is published, will reflect in maven central as per the usual SLA. Will do a review later today/tomorrow.

Thanks @nikhilnanivadekar for cutting the milestone release.

File: .github/workflows/javadoc.yml
Comment (nikhilnanivadekar) [reply .github/workflows/javadoc.yml:37]: if you have verified that the javadocs look ok, We should be fine 😄

File: missing-dependency/missing-dependency.iml
Comment (nikhilnanivadekar) [reply missing-dependency/missing-dependency.iml:3]: this should ideally auto-update. But the change is ok.

File: pom.xml
Comment (nikhilnanivadekar) [reply pom.xml:560]: We are using `<version>3.4.1</version>` for maven javadoc plugin. Do we need to upgrade this? It does not smell right that we are linking the 1.8 API docs in Javadoc but building on 11.

File: pom.xml
Comment (nikhilnanivadekar) [reply pom.xml:572]: I think this has to do with the javadoc plugin version. Also the delimiter might have changed: https://maven.apache.org/plugins/maven-javadoc-plugin/examples/exclude-package-names.html

**Review by nikhilnanivadekar — CHANGES_REQUESTED** [review]

File: pom.xml
Comment (prathasirisha) [reply pom.xml:572]: Delimiter has not changed. From the link you referenced: 

_If more than one package is to be excluded, each package name must be separated either with a colon (':'), a comma (',') or a semicolon (';')._

File: pom.xml
Comment (prathasirisha) [reply pom.xml:560]: I started out with 3.5.0, which is the latest version available but rolled back the change when I was running into build errors to eliminate any issues stemming from this plugin version upgrade. 

I can update this to 3.5.0 as it changes nothing for this PR and we will be on the latest version too.

Comment (prathasirisha) [reply pom.xml:560]: I agree it doesn't smell right that we are using 1.8 as the source. One option that I didnt explore is creating an issue [here](https://issues.apache.org/jira/projects/MJAVADOC/summary). I didnt see an issue with our exact problem, so that may be worth creating. I am open to other ideas as well.

**Comment by nikhilnanivadekar** [comment]
You can raise an issue. Please mention everything that you have tried.

File: pom.xml
Comment (prathasirisha) [reply pom.xml:560]: Created an issue - https://issues.apache.org/jira/browse/MJAVADOC-764

**Comment by nikhilnanivadekar** [comment]
@prathasirisha whats the latest on this? Do we want to pursue fixing the Javadoc or just merge? If you can verify that the Javadoc is fine, and spot check a few classes (EC & JDK) should we merge and proceed?

**Pull request closed without merging**
